src/main/java/bio/terra/cli/context/resources/AiNotebook.java

```diff
@@ -0,0 +1,320 @@
+package bio.terra.cli.context.resources;
+
+import bio.terra.cli.command.exception.UserActionableException;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.Resource;
+import bio.terra.cli.context.utils.Printer;
+import bio.terra.cli.service.GoogleAiNotebooks;
+import bio.terra.cli.service.WorkspaceManagerService;
+import bio.terra.cloudres.google.notebooks.InstanceName;
+import bio.terra.workspace.model.GcpAiNotebookInstanceAttributes;
+import bio.terra.workspace.model.GcpAiNotebookInstanceResource;
+import bio.terra.workspace.model.ResourceDescription;
+import bio.terra.workspace.model.ResourceType;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.api.services.notebooks.v1.model.Instance;
+import java.io.PrintStream;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@JsonDeserialize(builder = AiNotebook.AiNotebookBuilder.class)
+public class AiNotebook extends Resource {
+  private static final Logger logger = LoggerFactory.getLogger(AiNotebook.class);
+
+  public final String projectId;
+  public final String instanceId;
+  public final String location;
+  public final String machineType;
+  public final String postStartupScript;
+  public final Map<String, String> metadata;
+  public final String vmImageProject;
+  public final String vmImageFamily;
+  public final String vmImageName;
+  public final String containerRepository;
+  public final String containerTag;
+  public final String acceleratorType;
+  public final Long acceleratorCoreCount;
+  public final Boolean installGpuDriver;
+  public final String customGpuDriverPath;
+  public final Long bootDiskSizeGb;
+  public final String bootDiskType;
+  public final Long dataDiskSizeGb;
+  public final String dataDiskType;
```

Comment:
I actually don't think this class should contain all of these fields, or duplicate what's in the cloud for all of them.
There's a difference between the creation parameters and the business object used to represent the cloud resource. We only need these for the creation request and should only have them on that API. Conversely, there may be somethings about a resource (or other business object) we don't know until after it's created, e.g. the `Workspace.id`.
Also these can be modified by the user and by caching them here, we're signing ourselves up to keep them in sync, which we don't really have a need to do.

Reply:
I think this is addressed as part of the second pass to handle serialization. The `serialization.command.createupdate` package has objects that represent creation/update-time parameters.